hack/update-proto.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+script_dir=$(dirname $0)
+
+# Vendor dependencies, needed by go-to-protobuf
+go mod vendor
+
+go-to-protobuf \
+  --output-base=. \
+  --go-header-file "$script_dir/boilerplate.go.txt" \
+  --proto-import vendor \
+  --packages +./pkg/apis/topology/v1alpha1=v1alpha1 \
+  --keep-gogoproto=false \
+  --apimachinery-packages "-k8s.io/apimachinery/pkg/api/resource"
+
+# Hack to get the go_package option right
+sed s',go_package =.*,go_package = "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1";,' \
```

Comment:
@marquiz could you please clarify once again why do we need to replace go_package to full canonical path in proto file, since
proto generator puts only package name
https://github.com/kubernetes/code-generator/blob/master/cmd/go-to-protobuf/protobuf/generator.go#L50
I don't know is it reasonable or just a lack of functionality in code-generator.

Reply:
> @marquiz could you please clarify once again why do we need to replace go_package to full canonical path in proto file

This (from ithub.com/golang/protobuf/protoc-gen-go v1.4.3)

```
WARNING: Deprecated use of 'go_package' option without a full import path in "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1/generated.proto", please specify:
	option go_package = "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1";
A future release of protoc-gen-go will require the import path be specified.
```

Any newer versions of protoc-gen-go (e.g. all versions of google.golang.org/protobuf/cmd/protoc-gen-go) will just bluntly error out without that hack.



> I don't know is it reasonable or just a lack of functionality in code-generator.

I think it's just code-generator not being up-to-date. I guess patches are welcome there, too 😄